app/assets/javascripts/discourse-common/addon/resolver.js

```diff
@@ -85,252 +66,255 @@ export function buildResolver(baseName) {
       })) {
         if (fullName === key) {
           deprecated(`${key} was replaced with ${value}`, { since: "2.6.0" });
-          return value;
+          fullName = value;
         }
       }
 
+      this._uncorrectedValues ??= Object.create(null);
+
+      let normalized = super._normalize(fullName);
+
+      // This is code that we don't really want to keep long term. The main situation where we need it is for
+      // doing stuff like `controllerFor('adminWatchedWordsAction')` where the real route name
+      // is actually `adminWatchedWords.action`. The default behavior for the former is to
+      // normalize to `adminWatchedWordsAction` where the latter becomes `adminWatchedWords.action`.
+      // While these end up looking up the same file ultimately, they are treated as different
+      // items and so we can end up with two distinct version of the controller!
```

Comment:
I think this is self-explanatory, but let me know if it needs more information.

Reply:
The comment makes sense, but can we add a clear deprecation notice here with a suggested remedy?

So if someone uses

```
controllerFor('adminWatchedWordsAction')
```
and it resolves to `adminWatchedWords.action`, then we should print something like

> DEPRECATION: `adminWatchedWordsAction` was automatically corrected to `adminWatchedWords.action`. This auto-correction will be removed in the future. Use `adminWatchedWords.action` instead.

I see there is a deprecation notice a few lines lower, but it seems to only fire when there is a genuine clash?

Reply:
So unfortunately, it's actually a bit of the reverse of this. Basically `adminWatchedWordsAction` and `adminWatchedWords.action` resolve to `admin-watched-words-action`. What we really want is to resolve to `admin-watched-words/action`. I think I may have to work on this a bit more to make sure the slashed version is preferred if it exists.

Reply:
Could we add a list of aliases to the resolver that would allow requiring files from new places (with a deprecation warning) using old paths? That's would keep the mess in a single spot in code 😉

Reply:
Hmm... One other issue is that there is a precedence for using incorrect references in controllerFor and the like. For instance, there the following code: `this.controllerFor("admin-dashboard-general")` should be referencing it as `admin.dashboard.general`. That this is the correct route name can clearly be seen in the admin route map. If we relocate to the correct location, the dashed version stop working. We can change this in our own code, but any plugins referencing it this way will break.

Reply:
Another example is "adminWatchedWordsAction" which should be "adminWatchedWords.action".

Reply:
For now, I've decided to disable the deprecation notice. We definitely do want to deprecate this path, but there's more discussion necessary.